docs/ha-considerations.md

```diff
@@ -0,0 +1,332 @@
+# High Availability Considerations
+
+## Overview
+
+When setting up a production cluster, high availability (the cluster's ability to remain operational even if some control plane or worker nodes fail) is usually a requirement. For worker nodes, assuming that there are enough of them, this is part of the very cluster functionality. However redundancy of control plane nodes and `etcd` instances needs to be catered for when planning and setting up a cluster.
+
+`kubeadm` supports setting up of multi control plane and multi `etcd` clusters (see [Creating Highly Available clusters with kubeadm](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/high-availability/) for a step-by-step guide). Still there are some aspects to consider and set up which are not part of Kubernetes itself and hence not covered in the project documentation. This document provides some additional information and examples useful when planning and bootstrapping HA clusters with `kubeadm`.
+
+<!-- TOC -->
```

Comment:
I think we can move this before overview

Reply:
Maybe not if there's already the disclaimer on top? Would IMO not quite fit anymore.

Reply:
I just find weird to see the topic list not at the beginning of the doc, but it is only a nit

Reply:
That seems a matter of taste to me. I don't have any strong feelings here, so if you guys would like it changed I'm fine with it. In that case please give me an indication how you would like it. 

As of now the structure is: main header, disclaimer, then section Overview with a brief introduction text and the TOC, then the rest. I find this OK, but there's surely various other ways of structuring it.